src/audio/volume/volume.h

```diff
@@ -36,15 +36,17 @@ struct sof_ipc_ctrl_value_chan;
 
 #if defined(__XCC__)
 # include <xtensa/config/core-isa.h>
+#if CONFIG_VOLUME_SIMD_TOOLCHAIN_SELECT
 # if XCHAL_HAVE_HIFI4
-#  define VOLUME_HIFI4
+#  define CONFIG_VOLUME_HIFI4    1
 # elif XCHAL_HAVE_HIFI3
-#  define VOLUME_HIFI3
+#  define CONFIG_VOLUME_HIFI3    1
 # else
-#  define VOLUME_GENERIC
+#  define CONFIG_VOLUME_GENERIC  1
 # endif
+#endif
 #else
-# define VOLUME_GENERIC
+#  define CONFIG_VOLUME_GENERIC  1
```

Comment:
We don't have to keep duplicating this `#ifdef` logic in every single C file. I just submitted #8787 which is equivalent to the current state of this PR but without this duplication in every C file.